### ISSUE TITLE
Impala query taking time in unregister query or on Client Fetch Wait time

### ISSUE THREAD

**Issue by jsj051** [Automattic/php-thrift-sql#29]
look this url:https://issues.apache.org/jira/browse/IMPALA-2355

**Comment by xyu** [comment]
Can you provide an example / sample code that is causing this wait? In theory this library will continuously poll Impala for state and start fetching as soon as it's ready.

**Comment by xyu** [comment]
Ok looking at the above ticket we have:

```
Rows available: 1.16s (440ms)
First row fetched: 1.76s (594ms)
Unregister query: 15.0m (15.0m)
```

So the time between when the results were available and when the data was fetched was ~600ms which is the result of the exponential sleeper function here:

https://github.com/Automattic/php-thrift-sql/blob/master/src/ThriftSQL/Utils/Sleeper.php

I think ideally we could use a Gaussian decay for the amount of time to delay fetch events and tune it so that we fetch more frequently for queries that take say less then 10s or something but this delay does not seem that big of a deal overall.

More importantly I guess this issue and that ticket was concerned with the 15 minutes taken to get to `Unregister query`. That was because we didn't automatically closed the query handle after we we done fetching thus it waited until timeout or until the PHP processes exited. With #33, or more specifically 0f03c44d60839aa5a74b2fd4de6ade085727e544 this was fixed and a new `close()` method exposed so I'm closing this issue.
